src/ARCExpect/ValidationFunctions.fs

```diff
@@ -152,6 +152,18 @@ module Validate =
             |> ErrorMessage.ofIParam $"does not exist"
             |> Expecto.Tests.failtestNoStackf "%s"
 
+        /// <summary>
+        /// Validates if all elements in the given IParam collection satisfy the projection function.
+        /// </summary>
+        /// <param name="projection">A function that evaluates to true if the element satisfies the requirements.</param>
+        /// <param name="paramCollection">The IParam collection to validate.</param>
+        static member forAll (projection : #IParam -> bool) (paramCollection : #seq<#IParam>) =
+            match Seq.forall projection paramCollection with
+            | true  -> ()
+            | false ->
+                ErrorMessage.ofIParamCollection $"does not exist" paramCollection
```

Comment:
"does not exist" is not the correct error message here i think. since this should check wether the projection is true for all values, the same message as `ValueSatisfiesPredicate` ("is invalid) makes more sense.

However, there are 2 more issues here:
- please rename to `ValuesSatisfyPredicate`, as that is the name of the respective function for a single `IParam`. That way, it becomes `Validate.ParamCollection .ValuesSatisfyPredicate` ("Validate (for a) ParamCollection (that the) values satisfy (the) predicate")
- the error message seems to make even less sense to me combined with the new `ErrorMessage.ofIParamCollection`, since that uses the head of the sequence. Imagine the second value does not satisfy the predicate - but the error message uses the first - this is very confusing.

Reply:
It's just not about the Value but about the whole IParam. So the naming should be `ParamsSatisfyPredicate`.

To the second point: What would you think about reformulating the function so that it does not use `Seq.forAll` but iterates through the seq so that we can catch the failing IParams and return them?

Reply:
> It's just not about the Value but about the whole IParam

Ideally we would have all 3 variants for `Param` and `ParamCollection`.

Thinking again about this, i guess the names of the colelction versions should be a little different.


My suggestions would be

- `Param.SatisfiesPredicate`
- `Param.TermSatisfiesPredicate`
- `Param.ValueSatisfiesPredicate`
- `ParamCollection.ItemsSatisfyPredicate`
- `Param.TermsSatisfyPredicate`
- `Param.ValueySatisfyPredicate`